### ISSUE TITLE
Should accessibility tests ignore subviews of `UIButton`?

### ISSUE THREAD

**Issue by brennantaylor** [linkedin/LayoutTest-iOS#13]
When a `UIButton` is under LayoutTest, the subviews trip the automatic accessibility checks.

The ScreenReader seem to understand that the button is a button and doesn't recurse into the subviews.

I also don't want to disable the accessibility tests for the button in question because I do want the test to fail I don't set a label on the button since I'm controlling it a little more precisely.

So right now I'm resorting to the following in my tests:
```swift
for subview in view.subviews {
    self.viewsAllowingAccessibilityErrors.addObject(subview)
}
```

I'm wondering if I'm missing something because this seems odd.

**Comment by plivesey** [comment]
Hey @brennantaylor,
Somehow I missed the notifications for this. I didn't see any emails about this. Sorry about that.

**Comment by plivesey** [comment]
I'm a bit confused by this. You have a UIButton which has other subviews which are also UIButtons?

Possibly, we could add something like viewClassesAllowingAccessibilityErrors? But wondering if this is a normal usecase or if you're work around is correct.

**Comment by brennantaylor** [comment]
So this can happen if you add your own subviews to a UIButton. I don't think you can change the trait on the UIButton so the screen reader will always just treat it as a button and not inspect subviews.

I also see intermittent accessibility failures for the standard subviews (titleLabel and imageView for instance) inside the `UIButton`.

**Comment by plivesey** [comment]
Whats the failure that you get? Are you adding UIControls as subviews?

**Comment by plivesey** [comment]
I believe this is fixed in #18. We're ignoring all tests on subviews of buttons because the internals that apple provides sometimes breaks other tests.

Check out the latest on master. Going to close for now (will be releasing a new version to cocoapods soon).

**Comment by brennantaylor** [comment]
That did fix my issue. Thanks!
